src/Altinn.Platform/Altinn.Platform.Storage/Storage/Helpers/OrgDataContext.cs

```diff
@@ -0,0 +1,95 @@
+using System;
+using Altinn.Platform.Storage.Configuration;
+using Microsoft.Azure.KeyVault;
+using Microsoft.Azure.KeyVault.Models;
+using Microsoft.Extensions.Options;
+using Microsoft.WindowsAzure.Storage;
+using Microsoft.WindowsAzure.Storage.Auth;
+using Microsoft.WindowsAzure.Storage.Blob;
+
+namespace Altinn.Platform.Storage.Helpers
+{
+    /// <summary>
+    /// Contaxt to manage the organisations storage client
+    /// </summary>
+    public class OrgDataContext: IDisposable
+    {
+        /// <summary>
+        /// Gets the application owner blog container
+        /// </summary>
+        public CloudBlobContainer OrgBlobContainer { get; private set; }
+
+        private readonly AzureStorageConfiguration _storageConfiguration;
+
+        /// <summary>
+        /// Creates an instance of a <see cref="OrgDataContext"></see>
+        /// </summary>
+        /// <param name="org">Application owner name</param>
+        /// <param name="storageConfiguration">the storage configuration for azure blob storage</param>
+        public OrgDataContext(string org, AzureStorageConfiguration storageConfiguration)
+        {
+            _storageConfiguration = storageConfiguration;
+            OrgBlobContainer = GetCloudBlobContainer(org);
+        }
+
+        private CloudBlobContainer GetCloudBlobContainer(string org)
+        {
+            string containerName = string.Format(_storageConfiguration.OrgStorageContainer, org, Startup.EnvironmentName);
+            CloudBlobClient blobClient = GetCloudBlobClient(org);
+            if (blobClient != null)
+            {
+                return blobClient.GetContainerReference(containerName);
+            }
+
+            return null;
+        }
+
+        private CloudBlobClient GetCloudBlobClient(string org)
+        {
+            try
+            {
+                string secretUri = string.Format(_storageConfiguration.OrgKeyVaultURI, org, Startup.EnvironmentName);
+                string storageAccount = string.Format(_storageConfiguration.OrgStorageAccount, org, Startup.EnvironmentName);
+                string sasDefinition = string.Format(_storageConfiguration.OrgSasDefinition, org, Startup.EnvironmentName);
+
+                KeyVaultClient kv = Startup.PlatformKeyVaultClient;
```

Comment:
Consider creating a client per each application owner

Reply:
Short term solution ended up being setting Data and InstanceRepository to Transient.